question-servers/freeform.js

```diff
@@ -199,7 +217,7 @@ module.exports = {
             const courseErr = new Error('Invalid state before ' + phase + ': ' + checkErr);
             courseErr.fatal = true;
             courseErrs.push(courseErr);
-            return callback(null, courseErrs, data, '', fileData);
+            return callback(null, courseErrs, isGetDependencies ? elementDependencies : data, '', fileData);
```

Comment:
No, don't change, I think you are right. The growing-list-of-args is horrible. Collapsing them all back to a single arg would be even better :-)